test/cpp_api_parity/torch_nn_modules.py

```diff
@@ -1,193 +1,12 @@
+from collections import namedtuple
 import torch
 
-from cpp_api_parity import TorchNNModuleMetadata
+TorchNNModuleMetadata = namedtuple(
+    'TorchNNModuleMetadata',
+    [
+        'cpp_sources',
```

Comment:
When does one need to provide a `cpp_sources`?

Reply:
We only need to provide it for `SampleModule` and `sample_functional` (since their C++ implementation is not available in libtorch)

Reply:
Do you see us adding anything to this named tuple in the future? We really don't need a named tuple if there is only going to be a single argument.

Also, it would be nice to put your comment about that into the code (we only need to provide cpp_sources if we're testing an out-of-tree module and this is only done to test the testing harness)

Reply:
I changed `module_metadata_map` to `module_cpp_sources` and make it just be a dict of module name -> cpp sources (same for `functional_metadata_map` as well), and also added a comment for when we need to provide cpp sources :D